File: mpp-proiect-repository-saboalina-Lab3Java/src/main/java/domain/validators/BiletValidator.java
package domain.validators;

import domain.Bilet;

public class BiletValidator implements Validator<Bilet>{
    @Override
    public void validate(Bilet entity) throws ValidationException {
        StringBuilder err = new StringBuilder();
        if (entity.getId() <= 0)
            err.append("id bilet invalid\n");
        if(entity.getNumeClient().length()==0)
            err.append("nume client invalid\n");
        if(entity.getNumeTuristi().length()==0)
            err.append("nume turisti invalid\n");
        if(entity.getAdresaClient().length()==0)
            err.append("adresa client invalida\n");
        if (entity.getNrLocuri() <= 0)
            err.append("nr locuri invalid\n");
        if (entity.getZborId() <= 0)
            err.append("id zbor invalid\n");
        if(err.length() > 0)
            throw new ValidationException(err.toString());

    }
}

File: C#/Laborator3/domain/validators/BiletValidator.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Laborator4.domain.validators
{
    class BiletValidator : IValidator<Bilet>
    {
        public void Validate(Bilet e)
        {
            StringBuilder err = new StringBuilder();
            if (e.id <= 0)
                err.Append("id bilet invalid\n");
            if (e.numeClient.Length == 0)
                err.Append("nume client bilet invalid\n");
            if (e.numeTuristi.Length == 0)
                err.Append("nume turisti bilet invalid\n");
            if (e.adresaClient.Length == 0)
                err.Append("adresa client bilet invalida\n");
            if (e.nrLocuri <= 0)
                err.Append("nr locuri bilet invalid\n");
            if (e.zborId <= 0)
                err.Append("id zbor bilet invalid\n");
            if (err.Length > 0)
                throw new ValidationException(err.ToString());
        }
    }
}
